core/templates/pages/community-dashboard-page/question-opportunities/question-opportunities.directive.ts

```diff
@@ -105,11 +113,73 @@ angular.module('oppia').directive('questionOpportunities', [
           };
 
           ctrl.onClickSuggestQuestionButton = function(skillId) {
-            var opportunity = getOpportunity(skillId);
-            var question =
+            const modalInstance = $uibModal.open({
+              templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                '/pages/topic-editor-page/modal-templates/' +
+                'select-skill-and-difficulty-modal.template.html'),
+              backdrop: true,
+              controller: [
+                '$scope', '$uibModalInstance', 'DEFAULT_SKILL_DIFFICULTY',
+                'MODE_SELECT_DIFFICULTY', 'SkillDifficultyObjectFactory',
+                function($scope, $uibModalInstance, DEFAULT_SKILL_DIFFICULTY,
+                    MODE_SELECT_DIFFICULTY, SkillDifficultyObjectFactory) {
+                  const init = function() {
+                    $scope.instructionMessage = (
+                      'Select the skill(s) to link the question to:');
+                    $scope.currentMode = MODE_SELECT_DIFFICULTY;
+                    $scope.linkedSkillsWithDifficulty =
+                      [SkillDifficultyObjectFactory.create(
+                        skillId, '', DEFAULT_SKILL_DIFFICULTY)];
+                    SkillBackendApiService.fetchSkill(skillId)
```

Comment:
I think this should only be accessible to admins (not sure!) can you please check?

Reply:
Ah, you may be right. I believe [this](https://github.com/oppia/oppia/blob/develop/core/controllers/skill_editor.py#L117) is the controller method, and there is an edit ACL annotation. 

Does there need to be edit ACLs if it's just fetching a skill?

Also, how exactly do template URLs get routed to a particular controller method? Is there a mapping somewhere? Or does the framework handle that automatically somehow?

Reply:
> Does there need to be edit ACLs if it's just fetching a skill?

I think if we need to share this with contributors and in that case, it should be accessible to logged users @aks681, will it be fine to change the ACL for skillFetch controller? 


> how exactly do template URLs get routed to a particular controller method?

Yeah it''s defined in `main.py`

Reply:
Maybe add another controller for fetchReadOnlySkill? Since the URL is, I think, used for PUT requests as well. Although, you can just edit the acl for get(), not sure if we do it like that anywhere else, where only one part of the handler is restricted and the other is freely accessible.

I think it's fine to add another controller like ConceptCardHandler and return the skill there. Is that fine?

Reply:
>  not sure if we do it like that anywhere else, where only one part of the handler is restricted and the other is freely accessible.

@aks681 I think we do it in most of the places like explorationDataHandler is accessible to any user for GET request but PUT request is restricted to can_edit_exploration.

Reply:
Just to be clear, all this needs to be done on a case-by-case basis -- don't blindly apply a general pattern here. It's fine to treat each request handler (get/post/put/etc.) independently, just make sure the permissions are correct for each handler. Thanks!

Reply:
The get function for EditableSkillDataHandler also returns grouped_skill_summaries, though I think it's fine since it's just the summaries with the corresponding topic name which the normal user can access anyway.
So, yeah I think you can change the decorator for just the get() function and use it.

Reply:
Cool, I've updated the ACL to openaccess for the get method in core/controllers/skill_editor.py.